Source/Controllers/Other/SecureBookmarkData.swift

```diff
@@ -0,0 +1,65 @@
+//
+//  SecureBookmarkData.swift
+//  Sequel Ace
+//
+//  Created by James on 7/12/2020.
+//  Copyright © 2020 Sequel-Ace. All rights reserved.
+//
+
+import Foundation
+
+final class SecureBookmarkData: NSObject {
+    let bookmarkData: Data
+    private let options: Double
+    private let bookmarkURL: URL
+
+    init(data: Data, options: Double, url: URL) {
+        bookmarkData = data
+        self.options = options
+        bookmarkURL = url
+        super.init()
+    }
+}
+
+extension SecureBookmarkData: NSCoding, NSSecureCoding {
+    static var supportsSecureCoding: Bool {
+        return true
+    }
+
+    // MARK: NSCoding Implementation
+
+    enum Keys: String {
+        case bookmarkData = "BookmarkData"
+        case options = "Options"
+        case bookmarkURL = "BookmarkURL"
+    }
+
+    func encode(with coder: NSCoder) {
+        if #available(OSX 10.13, *) {
+            // For NSSecureCoding
+            coder.encode(bookmarkData as NSData, forKey: Keys.bookmarkData.rawValue)
+            coder.encode(NSNumber(value: options), forKey: Keys.options.rawValue)
+            coder.encode(bookmarkURL as NSURL, forKey: Keys.bookmarkURL.rawValue)
+        } else {
+            // For NSCoding
+            coder.encode(bookmarkData, forKey: Keys.bookmarkData.rawValue)
+            coder.encode(options, forKey: Keys.options.rawValue)
+            coder.encode(bookmarkURL, forKey: Keys.bookmarkURL.rawValue)
+        }
+    }
+
+    convenience init?(coder: NSCoder) {
+        if #available(OSX 10.13, *) {
```

Comment:
What happens if someone upgrades from macOS < 10.13 to macOS >= 10.13? Will encoding/decoding break? As long as we don't crash, it's fine, but we shouldn't crash if a user updates their Mac (why they would wait so long to update I don't know but it's always possible). Or we could drop support for macOS 10.12 imo

Reply:
This is what I was testing with my 10.12 VM. The bookmarks are just marked as stale and user is required to re-request access. Doesn't crash.